### PR TITLE
Hierarchical discovery, SDK→Edge rename, and server shim removal

### PR DESCRIPTION
**Pull request by atsyplikhin**
## Summary

- **Rename `device-connect-sdk` → `device-connect-edge`** across the entire monorepo (package, imports, CI, Dockerfiles)
- **Remove server re-export shims** — `device-connect-server` no longer re-exports drivers, messaging, telemetry, or device modules; consumers must import from `device-connect-edge` directly
- **Add hierarchical discovery** — three new tools (`describe_fleet`, `list_devices`, `get_device_functions`) with small-fleet auto-expansion, fuzzy type filtering, pagination, and grouping
- **Centralize normalization** in `_normalize.py` — shared `flatten_device`, `fuzzy_filter_by_type`, `extract_status`, `aggregate_fleet`, `group_devices` helpers used by both `tools.py` and the MCP bridge
- **Enhance registry service** with `discovery/getDevice` (O(1) lookup) and `describeFleet` RPC handlers, plus ACL filtering on all discovery methods
- **Improve D2D discovery** — `PresenceAnnouncer`/`PresenceCollector` with adaptive rate publishing (2 Hz burst → 0.2 Hz steady), peer TTL, and burst-on-new-peer
- **Version bump to 0.2.2**

## Breaking changes

- `from device_connect_server import DeviceRuntime` and similar re-exports no longer work — use `from device_connect_edge import ...`
- Devices must register with `capabilities.functions` / `capabilities.events` schema (not `base` or `static`)

## Test plan

- [x] 404 edge unit tests pass (`packages/device-connect-edge`)
- [x] 126 agent-tools unit tests pass (`packages/device-connect-agent-tools`)
- [x] 49 normalization tests cover fuzzy matching, status extraction, aggregation edge cases
- [x] 29 RPC handler tests cover register, list, get, describe, heartbeat (happy + error paths + ACL)
- [x] 466-line hierarchical discovery integration test suite (`test_tools_hierarchical.py`)
- [x] Integration tests with Docker infrastructure (`docker-compose-itest.yml`)

### CODE REVIEW

File: packages/device-connect-agent-tools/device_connect_agent_tools/adapters/strands_agent.py [hunk 1]
```diff
@@ -56,7 +56,9 @@ async def prepare(self) -> Dict[str, Any]:
         from strands import Agent
         from strands.models import AnthropicModel
         from device_connect_agent_tools.adapters.strands import (
-            discover_devices,
```
Comment (kavya-chennoju) [inline packages/device-connect-agent-tools/device_connect_agent_tools/adapters/strands_agent.py:59]: minor: Can strands_agent be moved under adapters -> agents -> strands_agent.py

**Review by kavya-chennoju — APPROVED** [review]

**Review by soupat — APPROVED** [review]
I have added a few TODOs in the private repo. Approved and merged.
